src/Features/Core/Portable/Diagnostics/DefaultDiagnosticAnalyzerService.cs

```diff
@@ -135,11 +135,18 @@ bool IsSemanticAnalysisOn()
 
             private async Task AnalyzeForKindAsync(TextDocument document, AnalysisKind kind, CancellationToken cancellationToken)
             {
-                var diagnosticData = await GetDiagnosticsAsync(document, kind, cancellationToken).ConfigureAwait(false);
+                // collect high pri diagnostics first, followed by normal pri.
+                await AnalyzeForKindAsync(highPriority: true).ConfigureAwait(false);
+                await AnalyzeForKindAsync(highPriority: false).ConfigureAwait(false);
 
-                _service.RaiseDiagnosticsUpdated(
-                    DiagnosticsUpdatedArgs.DiagnosticsCreated(new DefaultUpdateArgsId(_workspace.Kind, kind, document.Id),
-                    _workspace, document.Project.Solution, document.Project.Id, document.Id, diagnosticData));
+                async Task AnalyzeForKindAsync(bool highPriority)
+                {
+                    var diagnosticData = await GetDiagnosticsAsync(document, kind, highPriority, cancellationToken).ConfigureAwait(false);
+
+                    _service.RaiseDiagnosticsUpdated(
+                        DiagnosticsUpdatedArgs.DiagnosticsCreated(new DefaultUpdateArgsId(_workspace.Kind, kind, document.Id),
+                        _workspace, document.Project.Solution, document.Project.Id, document.Id, diagnosticData));
```

Comment:
@mavasani  i'm assuming it's fine to report multiple sets of diagnostics for the same doc.